### PR TITLE
BR-2350 - Save BuyerId On Registration

### PR DESCRIPTION
**Pull request by dhiraj-toptal**

### CODE REVIEW

File: Observer/CustomerRegisterSuccess.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    ) {
+        $this->_coreSession = $coreSession;
+        $this->balanceBuyer = $balanceBuyer;
+    }
+
+    public function execute(Observer $observer)
+    {
+        try {
+            $buyerId = $this->_coreSession->getBalanceBuyerId();
+            $this->balanceBuyer->updateCustomerBalanceBuyerId($buyerId);
```
Comment (ynaxonBalance) [inline Observer/CustomerRegisterSuccess.php]: what if there is no buyer id in session?

File: Observer/CustomerRegisterSuccess.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        } catch (\Exception $e) {
+
```
Comment (ynaxonBalance) [inline Observer/CustomerRegisterSuccess.php]: error handling?

File: Observer/CustomerRegisterSuccess.php (hunk 1 above)
Comment (dhiraj-toptal) [reply Observer/CustomerRegisterSuccess.php]: Updated buyer only if buyer_id present

File: Observer/CustomerRegisterSuccess.php (hunk 2 above)
Comment (dhiraj-toptal) [reply Observer/CustomerRegisterSuccess.php]: Added log in the catch

**Review by ynaxonBalance — DISMISSED** [review]
Please add on webkul as well

**Review by raimonbal — APPROVED** [review]

**Pull request merged**